src/bft.py

```diff
@@ -4,6 +4,18 @@
 from typing import Iterable
 from tree import T
 
+def bf_order_D_L(D: deque([T]), L: list[int]):
+    N = deque([])
+    for t in D:
+        if(t!=None):
+            N.append(t.left)
+            N.append(t.right)
+            L.append(t.val)
+    if (N != deque([])):
+        bf_order_D_L(N, L)
```

Comment:
This is a rather odd way to do it. The queue you have in `D` works fine, so why create a new one for the next level? You can just append to `D` instead of `N`. I also want to point out that you are using recursion in line 15, and while that often isn't a problem, it very much can be for large data structures (which is why we try to avoid it in this exercise).